src/client/pages/Geo/GeoMap/hooks/useGeoMap.ts

```diff
@@ -0,0 +1,87 @@
+import { MutableRefObject, useEffect, useRef, useState } from 'react'
+
+import { useAppDispatch } from 'client/store'
+import { GeoActions } from 'client/store/ui/geo'
+import { useCountryIso, usePrevious } from 'client/hooks'
+import { getCountryBounds } from 'client/pages/Geo/utils/countryBounds'
+import { mapController } from 'client/utils'
+
+type Props = {
+  viewport: google.maps.LatLngBoundsLiteral | null
+  zoom: number
+}
+
+type Returned = {
+  ref: MutableRefObject<HTMLDivElement>
+  map: google.maps.Map
+}
+
+export const useGeoMap = (props: Props): Returned => {
+  const { viewport, zoom } = props
+
+  const dispatch = useAppDispatch()
+
+  const ref = useRef<HTMLDivElement>(null)
+  const [map, setMap] = useState<google.maps.Map>()
+
+  const countryIso = useCountryIso()
+  const previousCountryIso = usePrevious(countryIso, countryIso)
+
+  useEffect(() => {
+    if (!ref.current || map) return
+
+    const mapSetup = new window.google.maps.Map(ref.current, {
+      controlSize: 30,
+      // There needs to be a default center, otherwise the map does not render
+      center: { lat: 0, lng: 0 },
+      disableDefaultUI: true,
+      fullscreenControl: true,
+      fullscreenControlOptions: {
+        position: google.maps.ControlPosition.TOP_CENTER,
+      },
+      mapTypeControl: true,
+      mapTypeControlOptions: {
+        mapTypeIds: [google.maps.MapTypeId.ROADMAP, google.maps.MapTypeId.SATELLITE, google.maps.MapTypeId.HYBRID],
+        position: google.maps.ControlPosition.TOP_CENTER,
+        style: google.maps.MapTypeControlStyle.DROPDOWN_MENU,
+      },
+      mapTypeId: google.maps.MapTypeId.HYBRID,
+      rotateControl: true,
+      zoom,
+      zoomControl: true,
+      zoomControlOptions: {
```

Comment:
Natively it is only possible to position them, it doesn't seem possible to align them. Another option could be to have the zoom on the left center and the full screen button in the top left, like in this [example](https://developers.google.com/maps/documentation/javascript/examples/control-positioning) (swapping street view with full screen):

![image](https://github.com/openforis/fra-platform/assets/41337901/200e2d0f-3d81-4dd0-8f5d-6839f330ce5a)

Reply:
Sure. Will add them to the top right corner. And the Map/Satellite is now a dropdown. I changed it so it takes less space. Should I revert it back to the style in the example above? (Clickable Map | Satellite text instead of a dropdown).

Reply:
This is how it would look like with smaller buttons on the top right, a bigger margin for the side panel, and Map/Satellite buttons as they were before:
 
![image](https://github.com/openforis/fra-platform/assets/41337901/1e4b7cda-dcf6-48d8-ad46-3e1681f5a74a)

![image](https://github.com/openforis/fra-platform/assets/41337901/ca94eafe-63f3-4b67-93fe-9ef415873608)

Reply:
I agree the zoom buttons aren't necessary, so I removed them. I moved the other buttons, and added an updated video to the description. 
I ended up not using the dropdown for the map type because the size change doesn't apply to it and it would look off:
![image](https://github.com/openforis/fra-platform/assets/41337901/d1687ccb-0a47-4142-a928-243d6505ab1d)